Runtime/utils/ABI.cs

```diff
@@ -86,6 +86,48 @@ private static PreparedParam GetEncodeValue(object obj, string inputType)
                 {
                     value = new BigInteger(intValue);
                 }
+                else if (obj is BigInteger bInt)
```

Comment:
[SHOULD] This can be refactored using `switch` pattern:
```cs
BigInteger value = obj switch {
  string v => v.StartsWith("0x") ? BigInteger.Parse(v.RemoveHexPrefix(), System.Globalization.NumberStyles.AllowHexSpecifier, null) : BigInteger.Parse(v),
  BigInteger v => v,
  int v => new BigInteger(v),
  long v => new BigInteger(v),
  _ => throw new System.NotSupportedException(),
};
```

Reply:
The logic is stable, I will not change!

Reply:
Done !